msgpack-core/src/main/java/org/msgpack/core/buffer/MessageBuffer.java

```diff
@@ -202,11 +219,32 @@ private static MessageBuffer newMessageBuffer(byte[] arr, int off, int len)
         }
     }
 
+    /**
+     * Creates a new MessageBuffer instance backed by ByteBuffer
+     *
+     * @param bb
+     * @return
+     */
+    private static MessageBuffer newMessageBuffer(ByteBuffer bb)
+    {
+        checkNotNull(bb);
+        try {
+            // We need to use reflection to create MessageBuffer instances in order to prevent TypeProfile generation for getInt method. TypeProfile will be
+            // generated to resolve one of the method references when two or more classes overrides the method.
+            return (MessageBuffer) mbBBConstructor.newInstance(bb);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     public static void releaseBuffer(MessageBuffer buffer)
     {
         if (isUniversalBuffer || buffer.base instanceof byte[]) {
```

Comment:
2GB is not so huge, so we should think about it if we are going to support huge message pack based DataFrame in memory. 

The index of getXXX(index) is using int, so we already have a technical limit of the accessible buffer range, but this should not limit the underlying buffer size; if we use `long[]` we can create a MessageBuffer slice (2GB range) using on 16GB memory buffer.